itou/eligibility/models/geiq.py

```diff
@@ -37,19 +37,24 @@
 
 
 class GEIQEligibilityDiagnosisQuerySet(CommonEligibilityDiagnosisQuerySet):
-    def authored_by_prescriber_or_geiq(self, geiq):
+    def authored_by_prescriber_or_geiq(self, geiq=None, for_job_seeker=False):
+        filter = models.Q()
+        if not for_job_seeker:
+            filter = models.Q(author_geiq=geiq) | models.Q(author_prescriber_organization__isnull=False)
         return (
-            self.filter(models.Q(author_geiq=geiq) | models.Q(author_prescriber_organization__isnull=False))
+            self.filter(filter)
             # Ordering by created_at is sufficient, because GEIQ can’t create a GEIQEligibilityDiagnosis
             # when there exists a valid GEIQEligibilityDiagnosis from an authorized prescriber.
             .order_by("-created_at")
         )
 
-    def diagnoses_for(self, job_seeker, for_geiq=None):
+    def diagnoses_for(self, job_seeker, for_geiq=None, for_job_seeker=False):
         # Get *all* GEIQ diagnoses for given job seeker (even expired)
+        # When for_job_seeker=False, fetch only diagnoses from authorized prescribers, or
+        # from a specified GEIQ; otherwise fetch all diagnoses.
         return (
             self.filter(job_seeker=job_seeker)
-            .authored_by_prescriber_or_geiq(for_geiq)
+            .authored_by_prescriber_or_geiq(for_geiq, for_job_seeker)
```

Comment:
Pourquoi appeler `authored_by_prescriber_or_geiq` si on a `for_job_seeker is True` ?

Car dans `authored_by_prescriber_or_geiq` on filtre si c'est créé par le GEIQ ou un PH (au passage, merci la contrainte, car à aucun moment on ne s'assure que le GEIQ n'est pas `None`), et je ne trouve pas cohérent avec le nommage si l'on ne filtre pas dès lors que c'est pour un candidat.